main.go

```diff
@@ -243,6 +247,18 @@ func visit(url *url.URL) {
 	}
 }
 
+func createBody(body string) (io.ReadCloser, error) {
+	if strings.HasPrefix(body, "@") {
+		filename := body[1:]
+		f, err := os.Open(filename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open data file %s: %v", filename, err)
+		}
+		return f, nil
+	}
+	return ioutil.NopCloser(strings.NewReader(body)), nil
```

Comment:
The great irony is that behaviour is now considered wrong. NewRequest should't just close the thing you pass to it, and forces people to use NopCloser to work around it's behaviour.